### PR TITLE
fix(type): add default type feature flag

### PR DESCRIPTION
**Pull request by jnm2377**
Ref #3648 

**Changed**

- Includes default type with its own feature flag.

### CODE REVIEW

File: packages/components/src/globals/scss/_css--reset.scss [hunk 1]
```diff
@@ -199,3 +199,9 @@
     @include carbon--icons;
   }
 }
+
+@include exports('css--default-type') {
+  @if variable-exists(css--default-type) == false or $css--default-type==true {
```
Comment (asudoh) [inline packages/components/src/globals/scss/_css--reset.scss]: Style nit - Probably spaces around `==`?

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @jnm2377! BTW `strong` thing is intentional, right?

**Review by alisonjoseph — APPROVED** [review]
LGTM! Assuming moving the strong declaration from default type because thats where it was before this change?

**Review by emyarod — COMMENTED** [review]
looks good to me, and seems the snapshots need updates to pass CI

**Comment by jnm2377** [comment]
@alisonjoseph yes, the strong declaration was in the css reset before this change, so I left it in there.

@emyarod will update the snapshots 👍

File: packages/components/src/globals/scss/_css--reset.scss (hunk 1 above)
Comment (jnm2377) [reply packages/components/src/globals/scss/_css--reset.scss]: Updated 👍

**Pull request merged**